pkg/controllers/controller.go

```diff
@@ -59,22 +58,23 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		apis.NewLivingConditionSet(conditions.Active).Manage(conditionsAccessor).MarkTrue(conditions.Active)
 	}
 	// 5. Reconcile
-	if _, err := c.Controller.Reconcile(ctx, resource); err != nil {
+	result, err := c.Controller.Reconcile(ctx, resource)
+	if err != nil {
 		zap.S().Errorf("Controller failed to reconcile kind %s, %s",
 			resource.GetObjectKind().GroupVersionKind().Kind, err.Error())
-		return reconcile.Result{Requeue: true}, nil
+		return result, err
 	}
 	// 6. Update Status using a merge patch
 	if err := c.Status().Patch(ctx, resource, client.MergeFrom(persisted)); err != nil {
-		return reconcile.Result{}, fmt.Errorf("Failed to persist changes to %s, %w", req.NamespacedName, err)
+		return result, fmt.Errorf("Failed to persist changes to %s, %w", req.NamespacedName, err)
 	}
-	return reconcile.Result{RequeueAfter: c.Interval()}, nil
```

Comment:
Now that we don't call Interval() here, we can remove it from the interface.

Reply:
Not quite. We did remove it from the GenericController code in order to allow different controllers to return the reconcile.Result that they desire. This allows us to requeue some controllers every 5 seconds, and some not.

Reply:
Discussed offline